From 53a0ed0f206db05ad4dafd5ddd98fb99f4f35a42 Mon Sep 17 00:00:00 2001
From: Ben Wiederhake <BenWiederhake.GitHub@gmx.de>
Date: Fri, 26 May 2023 19:45:47 +0200
Subject: [PATCH 4/4] Browser: Permit IPC traffic dumping

---
 Userland/Applications/Browser/main.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Userland/Applications/Browser/main.cpp b/Userland/Applications/Browser/main.cpp
index a9c30e0f13..21229c33e6 100644
--- a/Userland/Applications/Browser/main.cpp
+++ b/Userland/Applications/Browser/main.cpp
@@ -122,6 +122,9 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     TRY(Core::System::unveil("/tmp/session/%sid/portal/webcontent", "rw"));
     TRY(Core::System::unveil("/tmp/session/%sid/portal/request", "rw"));
     TRY(Core::System::unveil("/tmp/session/%sid/portal/sql", "rw"));
+    if (getenv("DUMP_LIBIPC_TRAFFIC")) {
+        TRY(Core::System::unveil("/tmp/", "rwc"));
+    }
     TRY(Core::System::unveil("/home", "rwc"));
     TRY(Core::System::unveil("/res", "r"));
     TRY(Core::System::unveil("/etc/passwd", "r"));
-- 
2.43.0

